src/main/java/net/coderbot/iris/shaderpack/ConfigOptionParser.java

```diff
@@ -0,0 +1,299 @@
+package net.coderbot.iris.shaderpack;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import net.minecraft.util.Util;
+
+public class ConfigOptionParser {
+	//Regex for matching boolean options
+	//Match if or if not the line starts with anynumber of backslashes that are more than 2 ("//")
+	//Match 0 or more whitspace after the "//"
+	//Match the #define
+	//Match 1 whitespace after that
+	//Match any letter, number, or underscore name
+	//Match 0 or more whitespace after that
+	//Match any comments on the option after that
+	private static final Pattern BOOLEAN_OPTION_PATTERN = Pattern.compile("^(?<startingComment>//+)?\\s*(?<define>#define)\\s+(?<name>\\w+)\\s*(?<comment>(?<commentChar>//+)(?<commentContent>.*))?$");
+	//Regex for matching ifdef patterns for boolean options
+	//Match the ifdef or ifndef keyword
+	//match whitespace that must be there
+	//match a word that is the name of the keyword
+	private static final Pattern IFDEF_IFNDEF_PATTERN = Pattern.compile("^(?<ifdef>#ifdef|#ifndef)\\s+(?<name>\\w+)(?<other>.*)");
+	//Regex that matches for integer and float options
+	//Match 1 or more whitespace after #define
+	//match a word (name of the option). Put in parenthesis for grouping
+	//match 1 or more whitespace after the name
+	   //match a negative if there
+	   //match f or F float keyword
+	   //match 0-9
+	   //match a "." for floats
+	 //match 1 or more of the char group
+	//match 0 or more whitespace
+	//match if there is a comment following the line or not
+	private static final Pattern FLOAT_INTEGER_OPTION_PATTERN = Pattern.compile("^(?<define>#define)\\s+(?<name>\\w+)\\s+(?<value>-?[\\d.fF]+)\\s*(?<comment>(?<commentChar>//+)(?<commentContent>.*))?$");
+	//Regex that matches for only integers and not floats
+	//Same as above but in the char class we remove the float specific checks
+		//remove matching a "."
+		//remove matching a "f" or a "F"
+	private static final Pattern INTEGER_OPTION_PATTERN = Pattern.compile("^(?<define>#define)\\s+(?<name>\\w+)\\s+(?<value>-?\\d+)\\s*(?<comment>(?<commentChar>//+)(?<commentContent>.*))?$");
+	//Some shaderpacks like sildurs have #define directives that are named with the program name
+	//like #define gbuffers_textured
+	//optifine does not use these in their config so we will not as well
+	//TODO figure what optifine does with these program defined names and implement
+	private static final Set<String> IGNORED_PROGRAM_NAMES = Util.make(new HashSet<>(), (set) -> {
```

Comment:
Would recommend you use multi-line comments throughout this whole thing rather than so many single line comments

Reply:
Yea sorry about that, I'll change them to multi-line comments